src/Module/User/Import.php

```diff
@@ -0,0 +1,415 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2022, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Module\User;
+
+use Friendica\App;
+use Friendica\Core\Config\Capability\IManageConfigValues;
+use Friendica\Core\L10n;
+use Friendica\Core\PConfig\Capability\IManagePersonalConfigValues;
+use Friendica\Core\Protocol;
+use Friendica\Core\Renderer;
+use Friendica\Core\System;
+use Friendica\Core\Worker;
+use Friendica\Database\Database;
+use Friendica\Database\DBA;
+use Friendica\Database\DBStructure;
+use Friendica\Model\Photo;
+use Friendica\Model\Profile;
+use Friendica\Module\Response;
+use Friendica\Navigation\SystemMessages;
+use Friendica\Network\HTTPException;
+use Friendica\Object\Image;
+use Friendica\Security\PermissionSet\Repository\PermissionSet;
+use Friendica\Util\Profiler;
+use Friendica\Util\Strings;
+use Friendica\Worker\Delivery;
+use Psr\Log\LoggerInterface;
+
+class Import extends \Friendica\BaseModule
+{
+	const IMPORT_DEBUG = false;
+
+	/** @var App */
+	private $app;
+
+	/** @var IManageConfigValues */
+	private $config;
+
+	/** @var IManagePersonalConfigValues */
+	private $pconfig;
+
+	/** @var SystemMessages */
+	private $systemMessages;
+
+	/** @var Database */
+	private $database;
+
+	/** @var PermissionSet */
+	private $permissionSet;
+
+	public function __construct(PermissionSet $permissionSet, IManagePersonalConfigValues $pconfig, Database $database, SystemMessages $systemMessages, IManageConfigValues $config, App $app, L10n $l10n, App\BaseURL $baseUrl, App\Arguments $args, LoggerInterface $logger, Profiler $profiler, Response $response, array $server, array $parameters = [])
+	{
+		parent::__construct($l10n, $baseUrl, $args, $logger, $profiler, $response, $server, $parameters);
+
+		$this->app            = $app;
+		$this->config         = $config;
+		$this->pconfig        = $pconfig;
+		$this->systemMessages = $systemMessages;
+		$this->database       = $database;
+		$this->permissionSet  = $permissionSet;
+	}
+
+	protected function post(array $request = [])
+	{
+		if ($this->config->get('config', 'register_policy') != \Friendica\Module\Register::OPEN && !$this->app->isSiteAdmin()) {
+			throw new HttpException\ForbiddenException($this->t('Permission denied.'));
```

Comment:
Isn't that a too generic message for the end user? Shouldn't it be more like `Import not permitted because of Registration policy`